### PR TITLE
Modelling User and Sign Up Page

### PR DESCRIPTION
**Pull request by syahidazuri**

### CODE REVIEW

**Comment by syahidazuri** [comment]
Ready

File: app/assets/javascripts/users.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
```
Comment (saobv-2223) [inline app/assets/javascripts/users.js]: remove file that never used

File: app/assets/config/settings.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  name: 
+    max_length: 50
+  micropost:
+    max_length: 140
+  email:
+    max_length: 255
+    regex: !ruby/regexp '/\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i'
+  password:
+    min_length: 6
+
```
Comment (saobv-2223) [inline app/assets/config/settings.yml]:
```suggestion
    min_length: 6

```

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/users.scss]: remove file that never used

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
+    debugger
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
    @user = User.find_by id: params[:id]
    return if @user 
    
    flash[:danger] = "Messages"
    redirect_to some_path
```

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  end
+    
+  def new;
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  def new
```

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    @user = User.new
+  end 
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "Let's start your course registration"
+      redirect_to @user
+    else
+      render "new"
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
      render :new
```

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    end
+  end
+
+  private
+
+    def user_params
+      params.require(:user).permit(:name, :email, :password, :password_confirmation)
+    end
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  def user_params
    params.require(:user).permit(:name, :email, :password, :password_confirmation)
  end
```

File: app/models/user.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { email.downcase! }
+  validates :name, presence: true,length: { maximum:50 }
+  VALID_EMAIL_REGEX= /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: true
+  has_secure_password
+  validates :password, presence: true, length: { minimum: 6 }
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  validates :password, presence: true, length: { minimum: 6 }
  
  has_secure_password
```

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { email.downcase! }
+  validates :name, presence: true,length: { maximum:50 }
+  VALID_EMAIL_REGEX= /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  VALID_EMAIL_REGEX= /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
  
  validates :name, presence: true,length: { maximum:50 }
```
and move before_save under has_secure_password

File: app/views/users/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+<div class="row">
+	<aside class="dol-md-4">
+		<section class="user_info">
+      <h1>
+        <%= gravatar_for @user %>
+        <%= @user.name %>
+			</h1>
+		</section>
+	</aside>
+</div>
```
Comment (saobv-2223) [inline app/views/users/show.html.erb]: check indenting again

**Comment by syahidazuri** [comment]
Ready

**Pull request merged**